models/trackeritem.go

```diff
@@ -0,0 +1,15 @@
+package models
+
+// TrackerItem represents a remote tracker item
+// Staging area before pushing to work item
+type TrackerItem struct {
+	ID uint64 `gorm:"primary_key"`
+	// Version for optimistic concurrency control
+	Version int
```

Comment:
Seems more sensible to me to add 'updated_at' which is auto supported by Gorm instead of a specific version field.

The updated_at field can be used in 2 cases:

* Last-Modified: updated_at
* ETag: hash(updated_at)

The optimistic locking can then rely on comparing hash(o.updated_at) == ETag
(if optimistic locking is even required)